pkg/util/node.go

```diff
@@ -49,3 +56,67 @@ func IsNodeAddressTypeSupported(addrType corev1.NodeAddressType) bool {
 	}
 	return false
 }
+
+// GetCPUsReservedByNodeAnno gets reserved cpus by annotation of node.
+func GetCPUsReservedByNodeAnno(anno map[string]string) (corev1.ResourceList, error) {
+	rl := make(corev1.ResourceList)
+
+	if reserved, ok := anno[apiext.ReservedByNode]; ok {
+		cpuReservedByNode, qos := GetCPUsReservedByNode(reserved)
+		if qos == apiext.QoSLSE {
+			rl[corev1.ResourceCPU] = cpuReservedByNode
+		}
+	}
+
+	return rl, nil
+}
+
+func GetCPUsReservedByNode(reserved string) (resource.Quantity, apiext.QoSClass) {
+	cpuReservedByNode := *resource.NewMilliQuantity(0, resource.DecimalSI)
+	reservedObj := apiext.KoordReserved{}
+	qos := apiext.QoSNone
+
+	if err := json.Unmarshal([]byte(reserved), &reservedObj); err != nil {
+		klog.Errorf("Failed to unmarshal cpus reserved by node annotation. err=%v.\n", err)
+		return cpuReservedByNode, qos
+	}
+
+	if reservedcpu, ok := reservedObj.ReservedResources[corev1.ResourceCPU]; ok {
+		cpuReservedByNode = reservedcpu
+	}
+	if reservedObj.ReservedCPUs != "" {
+		if cpus, err := cpuset.Parse(reservedObj.ReservedCPUs); err == nil {
+			cpuReservedByNode = resource.MustParse(strconv.Itoa(cpus.Size()))
+		}
+	}
+
+	return cpuReservedByNode, reservedObj.QOSEffected
+}
+
+// RemoveNodeReservedCPUs filter out cpus that reserved by annotation of node.
+func RemoveNodeReservedCPUs(cpuSharePools []apiext.CPUSharedPool, cpusReservedByNodeAnno string) []apiext.CPUSharedPool {
```

Comment:
I think the function `RemoveNodeReservedCPUs` maybe just used by koordlet, move the function to the koordlet/util?